include/spdlog/sinks/daily_file_sink.h

```diff
@@ -104,6 +103,28 @@ class daily_file_sink final : public base_sink<Mutex>
     }
 
 private:
+    void init_filenames_q()
+    {
+        using details::os::path_exists;
+
+        filenames_q_ = details::circular_q<filename_t>(static_cast<size_t>(max_files_));
+        std::vector<filename_t> filenames;
```

Comment:
This vector can be avoided. Just pushback to the q  instead of accumulating in filenames.

Reply:
I think it will remove the latest log file when rotating if push back to the q directly.
I am confused whether It leads to an mistake that could not get the previews log files when service or process is restarted.
In this way it will never remove any previews files.
@gabime

Reply:
right